addons/mail/models/mail_thread.py

```diff
@@ -1880,6 +1880,24 @@ def _message_post_after_hook(self, message, msg_vals):
         using already-computed values instead of having to rebrowse things. """
         pass
 
+    @api.model
+    def _message_edit_content(self, message, body):
```

Comment:
Would make a classic api.multi and check access on document itself (and maybe just check that message.res_id = self.id, and raise a Value Error otherwise)